### ISSUE TITLE
linter: getting 'unknown field "settings"' error in overrides

### ISSUE THREAD

**Issue by nebrius** [oxc-project/oxc#21058]
### What version of Oxlint are you using?

1.57.0

### What command did you run?

oxlint  --type-aware

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```js
export default {
  overrides: [
    {
      settings: {},
    },
  ],
};
```

### What happened?

I got an error stating:

```
  × Failed to parse config from /Users/bryan/Projects/eslint-plugin-fast-import/oxlint.config.ts
  note: unknown field `settings`, expected one of `files`, `env`, `globals`, `plugins`, `jsPlugins`, `rules`
```

I expected this to behave like ESLint, which supports scoping settings to specific file paths.

For context, I'm the author of https://github.com/nebrius/eslint-plugin-fast-import, and I'm attempting to add support for Oxlint, but I'm struggling. My primary goal is to support monorepos, but I just can't seem to connect all the dots in Oxlint.

The core of the problem is that I need the ability for users to configure settings that controls computing import/export info, such as entry points expected to be imported by other packages, externally imported exports (aka entry points imported by third party systems, such as default exports in page.tsx/layouts.tsx/etc in Next.js), etc. This information must be shared across all rules, and so really is a plugin-wide setting, _but_ is also scoped to a package within a monorepo, not the entire codebase.

I would recommend that users create separate lint configs per package to work around this bug/limitation, but:
1. This approach is blocked by https://github.com/oxc-project/oxc/issues/19937
2. I have plans to implement monorepo-wide rules that necessitate a single top-level config that encompasses all packages (e.g. "is this entry point in a given package imported or not in other packages" and "does this entry point create a circular import with another package")
